examples/navier-stokes/navierstokes.c

```diff
@@ -647,7 +647,7 @@ int main(int argc, char **argv) {
   if (!icsfp)
       return CeedError(ceed, 1, "Function not found in the list");
   char str[256] = __FILE__":ICs";
-  strcat(str, problemtype);
+  PetscStrlcat(str, problemtype, 256);
```

Comment:
I'm not sure this is safe enough. Should we make it safer by stating that the maximum number of character copied into str should be no more than sizeof(str) - strlen(str)?

Reply:
PetscStrlcat makes sure that we don't go past the declared length of str.

Reply:
Note that you can use `sizeof(str)` in situations like this.